dist/pythonlibs/riotnode/riotnode/shell/gnrc.py

```diff
@@ -0,0 +1,67 @@
+"""Mixins for GNRC-related shell interaction
+
+Defines mixins for GNRC-related shell commands
+"""
+
+from .base import ShellInteraction
+
+
+class GNRCICMPv6Echo(ShellInteraction):
+    class Results(object):
+        def __init__(self):
+            self.replies = []
+            self.stats = {}
+            self.rtts = {}
+
+        def _add_reply(self, reply):
+            reply["seq"] = int(reply["seq"])
+            reply["ttl"] = int(reply["ttl"])
+            reply["rtt"] = int(reply["rtt"])
+            if "rssi" in reply:
+                reply["rssi"] = int(reply["rssi"])
+            self.replies.append(reply)
+
+        def _set_stats(self, stats):
+            stats["packet_loss"] = int(stats["packet_loss"])
+            stats["rx"] = int(stats["rx"])
+            stats["tx"] = int(stats["tx"])
+            self.stats.update(stats)
+
+        def _set_rtts(self, rtts):
+            rtts["min"] = int(rtts["min"])
+            rtts["avg"] = int(rtts["avg"])
+            rtts["max"] = int(rtts["max"])
+            self.rtts.update(rtts)
+
+    @ShellInteraction.check_term
+    def ping6(self, hostname, count=3, interval=1000, packet_size=4,
+              hop_limit=None, timeout=1000):
+        cmd = "ping6 {hostname} -c {count} -i {interval} " \
+              "-s {packet_size} -W {timeout}" \
+              .format(hostname=hostname, count=count, interval=interval,
+                      packet_size=packet_size, timeout=timeout)
+
+        if hop_limit is not None:
+            cmd += " -h {hop_limit}".format(hop_limit)
+
+        term = self.node.term
+        term.sendline(cmd)
+        res = GNRCICMPv6Echo.Results()
+
+        for c in range(count + 1):
+            res = term.expect([r"\d+ bytes from (?P<source>[0-9a-f:]+): "
+                               r"icmp_seq=(?P<seq>\d+) ttl=(?P<ttl>\d+) "
+                               r"(rssi=(?P<rssi>-?\d+) dBm )?"
+                               r"time=(?P<rtt>\d+.\d+) ms",
```

Comment:
Note to self: `rtt` can also be omitted (when `packet_size < sizeof(uint32_t)`).